File: app/src/main/java/za/co/woolworths/financial/services/android/util/frag_nav/tab_history/CurrentTabHistoryController.java
package za.co.woolworths.financial.services.android.util.frag_nav.tab_history;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import za.co.woolworths.financial.services.android.util.frag_nav.FragNavPopController;
import za.co.woolworths.financial.services.android.util.frag_nav.FragNavTransactionOptions;

public class CurrentTabHistoryController extends BaseFragNavTabHistoryController {
    public CurrentTabHistoryController(FragNavPopController fragNavPopController) {
        super(fragNavPopController);
    }

    @Override
    public boolean popFragments(int popDepth,
                                FragNavTransactionOptions transactionOptions) throws UnsupportedOperationException {
        return fragNavPopController.tryPopFragments(popDepth, transactionOptions) > 0;
    }

    @Override
    public void switchTab(int index) {
    }

    @Override
    public void onSaveInstanceState(@NonNull Bundle outState) {
    }

    @Override
    public void restoreFromBundle(@Nullable Bundle savedInstanceState) {
    }
}

File: app/src/main/java/za/co/woolworths/financial/services/android/util/nav/tabhistory/CurrentTabHistoryController.kt
package za.co.woolworths.financial.services.android.util.nav.tabhistory

import android.os.Bundle

import za.co.woolworths.financial.services.android.util.nav.FragNavPopController
import za.co.woolworths.financial.services.android.util.nav.FragNavTransactionOptions

class CurrentTabHistoryController(fragNavPopController: FragNavPopController) : BaseFragNavTabHistoryController(fragNavPopController) {

    @Throws(UnsupportedOperationException::class)
    override fun popFragments(popDepth: Int,
                              transactionOptions: FragNavTransactionOptions?): Boolean {
        return fragNavPopController.tryPopFragments(popDepth, transactionOptions) > 0
    }

    override fun switchTab(index: Int) {}

    override fun onSaveInstanceState(outState: Bundle) {}

    override fun restoreFromBundle(savedInstanceState: Bundle?) {}
}
